### PR TITLE
optional https port for s3

### PR DESCRIPTION
**Pull request by kmlebedev**
# What problem are we solving?

In kubernetes, traffic can be delivered through ingress, which itself allows you to do TLS and through the internal k8s service directly to the s3 api application and I would like to have additional https with TLS

It will also help to encrypt traffic if it needs to be sent past the ingress https://github.com/seaweedfs/seaweedfs/issues/4479

# How are we solving the problem?

optional https port for s3 api

# How is the PR tested?

local with param `-s3.port=8333 -s3.port.https=8433 -s3.key.file=/usr/local/share/ca-certificates/tls.key -s3.cert.file=/usr/local/share/ca-certificates/tls.crt`
```
seaweedfs-s3-1      | I0518 12:26:07.332791 s3.go:263 Start Seaweed S3 API Server 30GB 3.50  at https port 8433
seaweedfs-s3-1      | I0518 12:26:07.333679 s3.go:281 Start Seaweed S3 API Server 30GB 3.50  at http port 8333
```

local with param `-s3.port=8333 -s3.key.file=/usr/local/share/ca-certificates/tls.key -s3.cert.file=/usr/local/share/ca-certificates/tls.crt`
```
seaweedfs-s3-1      | I0518 12:26:07.333679 s3.go:281 Start Seaweed S3 API Server 30GB 3.50  at https port 8333
```

local with param `-s3.port=8333`
```
seaweedfs-s3-1      | I0518 12:26:07.333679 s3.go:281 Start Seaweed S3 API Server 30GB 3.50  at http port 8333
```

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/command/s3.go [hunk 1]
```diff
@@ -245,18 +247,37 @@ func (s3opt *S3Options) startS3Server() bool {
 	go grpcS.Serve(grpcL)
 
 	if *s3opt.tlsPrivateKey != "" {
-		glog.V(0).Infof("Start Seaweed S3 API Server %s at https port %d", util.Version(), *s3opt.port)
-		if s3ApiLocalListener != nil {
+		if *s3opt.portHttps == 0 {
+			glog.V(0).Infof("Start Seaweed S3 API Server %s at https port %d", util.Version(), *s3opt.port)
```
Comment (chrislusf) [inline weed/command/s3.go:270]: the message should be "http port"?

Comment (kmlebedev) [reply weed/command/s3.go:270]: No. Should be http because this message in blok with tls key exist.

**Comment by kmlebedev** [comment]
@chrislusf Hey, is this getting better?

File: weed/command/s3.go (hunk 1 above)
Comment (chrislusf) [reply weed/command/s3.go:270]:
```suggestion
			glog.V(0).Infof("Start Seaweed S3 API Server %s at http port %d", util.Version(), *s3opt.port)
```

File: weed/command/s3.go [hunk 2]
```diff
@@ -245,18 +247,37 @@ func (s3opt *S3Options) startS3Server() bool {
...
 					glog.Fatalf("S3 API Server Fail to serve: %v", err)
 				}
 			}()
 		}
-		if err = httpS.ServeTLS(s3ApiListener, *s3opt.tlsCertificate, *s3opt.tlsPrivateKey); err != nil {
-			glog.Fatalf("S3 API Server Fail to serve: %v", err)
-		}
-	} else {
+	}
+	if *s3opt.tlsPrivateKey == "" || *s3opt.portHttps > 0 {
```
Comment (chrislusf) [inline weed/command/s3.go:299]: not sure what is the intent here

Comment (kmlebedev) [reply weed/command/s3.go:299]: > not sure what is the intent here

@chrislusf The general idea is to run the s3 http server simultaneously on http and https different ports, while remaining backwards compatible when using a common port for http + TLS

Accordingly, this line allows you to start an http server without TLS on the "main" port, when TLS is already running on another https port

**Comment by kmlebedev** [comment]
@chrislusf ping

**Comment by chrislusf** [comment]
> @chrislusf Hey, is this getting better?

comments were not addressed yet.

**Comment by chrislusf** [comment]
> > comments were not addressed yet.
> 
> Ok. translated into resolved.

They are marked as resolved. but no actual changes made.

File: weed/command/s3.go (hunk 1 above)
Comment (kmlebedev) [reply weed/command/s3.go:270]: @chrislusf I didn’t successfully name the variable `portHttps`, but if it differs from 0 to zero, it means you need to run TLS on a separate port and in this code block only TLS is always started, so the message will always be https

logging on start
```
I0525 10:51:35.310073 s3.go:263 Start Seaweed S3 API Server 30GB 3.51 8e59d8fec at https port 443
I0525 10:51:35.310119 s3.go:281 Start Seaweed S3 API Server 30GB 3.51 8e59d8fec at http port 8080
```

**Comment by chrislusf** [comment]
I suspect you did not see my comments which were marked as resolved.

File: weed/command/s3.go (hunk 2 above)
Comment (kmlebedev) [reply weed/command/s3.go:299]: Did you see this comment ?

**Review by kmlebedev — COMMENTED** [review]
try finish

**Comment by kmlebedev** [comment]
> I suspect you did not see my comments which were marked as resolved.

It looks like I created a review and until I finished it all the messages were in the pending status

**Pull request merged**